File: frontend/lib/models/exchange_rate.dart
class ExchangeRate {
  final String currency;
  final double rate;

  ExchangeRate({
    required this.currency,
    required this.rate,
  });

  // Convert JSON to ExchangeRate object
  factory ExchangeRate.fromJson(Map<String, dynamic> json) {
    return ExchangeRate(
      currency: json['currency'] as String,
      rate: (json['rate'] as num).toDouble(),
    );
  }

  // Convert ExchangeRate object to JSON
  Map<String, dynamic> toJson() {
    return {
      'currency': currency,
      'rate': rate,
    };
  }
}

File: backend/models/ExchangeRate.js
const mongoose = require('mongoose');

const exchangeRateSchema = new mongoose.Schema({
    currency: {
        type: String,
        required: true,
    },
    rate: {
        type: Number,
        required: true,
    },
});

const ExchangeRate = mongoose.model('ExchangeRate', exchangeRateSchema);

module.exports = ExchangeRate;
